fsck/fsck.c

```diff
@@ -670,58 +670,72 @@ static int read_boot_region(struct exfat_blk_dev *bd, struct pbr **pbr,
 		return -ENOMEM;
 	}
 
+	if (main_sect)
+		bs_offset = BOOT_SEC_IDX;
+	else
+		bs_offset = BACKUP_BOOT_SEC_IDX;
+
 	if (exfat_read(bd->dev_fd, bs, sizeof(*bs),
-			bs_offset * bd->sector_size) != (ssize_t)sizeof(*bs)) {
+			bs_offset * sect_size) != (ssize_t)sizeof(*bs)) {
 		exfat_err("failed to read a boot sector\n");
 		ret = -EIO;
 		goto err;
 	}
 
 	if (memcmp(bs->bpb.oem_name, "EXFAT   ", 8) != 0) {
-		exfat_err("failed to find exfat file system.\n");
+		if (main_sect)
+			exfat_err("failed to find exfat file system\n");
 		goto err;
 	}
 
-	ret = boot_region_checksum(bd, bs_offset);
+	ret = boot_region_checksum(bd, bs_offset, sect_size);
```

Comment:
If bd->sector_size is no long used, we don't need to make struct bd as argument of boot_region_checksum(). just give bd->dev_fd as an agument ?

Reply:
Replacing bd with bd->dev_fd seems good. But I don't think the bd parameter is strange, and the sect_size parameter means Exfat's sector size.

Reply:
I have nothing to say.